opamp/config.go

```diff
@@ -57,3 +60,36 @@ func ParseConfig(configLocation string) (*Config, error) {
 
 	return &config, nil
 }
+
+// GetSecretKey returns secret key if set else returns empty string
+func (c Config) GetSecretKey() string {
+	if c.SecretKey == nil {
+		return ""
+	}
+
+	return *c.SecretKey
+}
+
+// CmpUpdatableFields compares updatable fields for equality
+func (c Config) CmpUpdatableFields(o Config) (equal bool) {
+	if !cmpStringPtr(c.AgentName, o.AgentName) {
+		return false
+	}
+
+	return cmpStringPtr(c.Labels, o.Labels)
+}
+
+func cmpStringPtr(p1, p2 *string) bool {
+	switch {
+	case p1 == nil && p2 == nil:
+		return true
+	case p1 == nil && p2 != nil:
```

Comment:
maybe something is happening in go that i don't understand here, but are these "fallthrough" cases necessary?

Reply:
So switch statements in go don't fallthrough by default like most languages do. You have to tell it to do so. Since all cases return false I added fallthrough. I could also just put `return false` in every case.

Reply:
@cpheps i mean what would happen if you completely removed those to 1 thing == nil cases

Reply:
Not sure I follow what what you're asking.

Reply:
cleared this up. just didn't understand exactly how fallthrough works in go